packages/chat-headless/src/ChatHeadlessImpl.ts

```diff
@@ -264,6 +266,11 @@ export class ChatHeadlessImpl implements ChatHeadless {
   }
 
   restartConversation() {
+    if (isChatEventClient(this.chatClient)) {
```

Comment:
while this is not the case for our current amazon connect use case, technically bot/agent client can be an http or event client. I think we should check if this is actually an agent client, then set to botClient (and if agent client is of event client type then we can reset)